pkgs/applications/version-management/git-and-tools/gitin/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, buildGoPackage, fetchFromGitHub, pkgconfig, libgit2 }:
+
+let
+  libgit2_0_27 = libgit2.overrideAttrs (oldAttrs: rec {
```

Comment:
Why do you override it?

Reply:
I get this error without it :

```sh
nix-build -A gitAndTools.gitin
# these derivations will be built:
#   /nix/store/c0ln4ca8gq1r0aya77ahdz84vik0raj1-gitin-0.2.3.drv
# building '/nix/store/c0ln4ca8gq1r0aya77ahdz84vik0raj1-gitin-0.2.3.drv'...
# unpacking sources
# unpacking source archive /nix/store/ljsn5lg2ybh2qkag1f8qr71rki3pljlf-source
# source root is source
# patching sources
# configuring
# unpacking source archive /nix/store/wdbn0qm3a3zmw53016cwh9ywyvhwc5h3-template-fb15b89
# unpacking source archive /nix/store/l9cr5h9cakr9vlwqvlann98liaqchsl0-units-f65c72e
# unpacking source archive /nix/store/1c04ch224a6bvkqvsv603y88jhaldwkm-color-daf2830
# unpacking source archive /nix/store/5djl5zmr5fhni1b33i6ia33knjnqyi1v-gia-0055649
# unpacking source archive /nix/store/6snfp9ksg58k7j2rvvgd3sq025dzqp32-gocui-c055c87
# unpacking source archive /nix/store/ggs3nd28qjj2mawfs6jlx4k2w08z5icd-timeago-027f403
# unpacking source archive /nix/store/pjbw04f8y98yccjwxahhamxfpcjv77yz-envconfig-0b417c4
# unpacking source archive /nix/store/lmwldqi5k0kkqz21wssm9y07h275zp7z-go-runewidth-14e809f
# unpacking source archive /nix/store/dcbmk28d2sy1kjjl220y7h4v3fakvvbv-termbox-go-38ba6e5
# unpacking source archive /nix/store/2gaynbl1kv9rx043a4mqxy9shc87iv54-fuzzy-d88f8cb
# unpacking source archive /nix/store/ljdq7xsgi3nnmpw2fjw5kfi124bzi45b-diffparser-7391f21
# unpacking source archive /nix/store/kbhz34cxjjzxv5alxascxmmvyg0cifjy-kingpin.v2-947dcec
# unpacking source archive /nix/store/rsx5370h8mr1xbsg43bn38sq7q9narpj-git2go.v27-6cc7d3d
# building
# runtime/cgo
# github.com/waigani/diffparser
# github.com/fatih/color/vendor/golang.org/x/sys/unix
# github.com/mattn/go-runewidth
# github.com/nsf/termbox-go
# github.com/sahilm/fuzzy
# github.com/justincampbell/timeago
# github.com/kelseyhightower/envconfig
# github.com/jroimartin/gocui
# github.com/alecthomas/template/parse
# github.com/fatih/color/vendor/github.com/mattn/go-isatty
# github.com/fatih/color/vendor/github.com/mattn/go-colorable
# github.com/alecthomas/units
# github.com/fatih/color
# github.com/isacikgoz/gia/editor
# github.com/isacikgoz/gitin/term
# github.com/alecthomas/template
# github.com/isacikgoz/gitin/prompt
# net
# gopkg.in/alecthomas/kingpin.v2
# gopkg.in/libgit2/git2go.v27
# # gopkg.in/libgit2/git2go.v27
# go/src/gopkg.in/libgit2/git2go.v27/git_dynamic.go:10:3: error: #error "Invalid libgit2 version; this git2go supports libgit2 v0.27"
#    10 | # error "Invalid libgit2 version; this git2go supports libgit2 v0.27"
#       |   ^~~~~
# builder for '/nix/store/c0ln4ca8gq1r0aya77ahdz84vik0raj1-gitin-0.2.3.drv' failed with exit code 1
# error: build of '/nix/store/c0ln4ca8gq1r0aya77ahdz84vik0raj1-gitin-0.2.3.drv' failed
```

Reply:
I also found this to be done in:
`pkgs/applications/version-management/gitlab/gitaly/default.nix`

Reply:
OK, that's acceptable. But, considering the fact we have now more then 1 package requiring this particular version of libgit2, I think it'd be proper to move it into `all-packages.nix` with an `overrideAttrs` just like here and in `gitaly/default.nix`.

If you do that, I'd make the PR have these commits:

```
maintainers: add kimat
libgit2_0_27: init at 0.27
gitaly: use ligit2 from all-packages.nix
gitin: init at 0.2.3
```

And, also make sure that the build of `gitaly` doesn't change **at all** - meaning you run `nix-build -A gitaly` (or whatever) and you get the binary from cache.nixos.org.

Reply:
I just added those changes and made sure these were passing :

```sh
nixpkgs-review pr 86070
nix-build -A gitAndTools.gitaly
nix-build -A gitaly
nix-build -A gitlab # depends on gitaly
```